### DISCUSSION TITLE
Limit RAM available for rootfs

### DISCUSSION THREAD

**Discussion by MaximShepelev** [ltsp/ltsp#477] [Q&A] [answered]
Hello,

It seems that when booting ltsp always sets the size of rootfs to half of total physical RAM

Case 1 [791581468 = 2*395790732 +4] (4KiB got lost somewhere :D)
```
ops@kub07:~$ free
              total        used        free      shared  buff/cache   available
Mem:      791581468     3279320   785266800     2334596     3035348   782410360
Swap:             0           0           0
ops@kub07:~$ df
Filesystem          1K-blocks     Used Available Use% Mounted on
udev                395760044        0 395760044   0% /dev
tmpfs                79158148     4012  79154136   1% /run
/run/initramfs/ltsp 395790732  2330528 393460204   1% /

```

Case 2 [10205272 = 2*5102636]
```
ops@msh-pxe-test-3:~$ free 
              total        used        free      shared  buff/cache   available
Mem:       10205272      396240     6734968     2591144     3074064     6992896
Swap:             0           0           0
ops@msh-pxe-test-3:~$ df
Filesystem          1K-blocks    Used Available Use% Mounted on
udev                  5071948       0   5071948   0% /dev
tmpfs                 1020528    2608   1017920   1% /run
/run/initramfs/ltsp   5102636 2588456   2514180  51% /
```

Is there a way to specify how much RAM is allocated for rootfs? I didn't find any according documentation nor did I find anything in code that is responsible for this weird behavior. 

We are using ltsp to bootstrap Kubernetes compute nodes into cluster and we want to be on the safe side and control how much RAM is available for rootfs and don't want to allocate even half of it.
Our use case is with host getting filled with Container images over time which are not cleared until **HighThresholdPercent** and **LowThresholdPercent** values are met. Those can be tuned accordingly to artificially cleaned congested RAM. However those parameters may vary based on total hosts total RAM and reaching those limits will lead to pod eviction which in turn would lead to freeing up RAM and stopping cleanup process creating a loop. 
In addition containers logs can fill fs with time which will make less RAM available for compute resources and less pods being scheduled.

**Comment by alkisg** [comment, marked as answer]
Hi, are you using IMAGE_TO_RAM?

Open /usr/share/ltsp/client/initrd-bottom/55-initrd-bottom.sh, and change this line: `re vmount -t tmpfs -o mode=0755 tmpfs "$tmpfs"`

to: `re vmount -t tmpfs -o mode=0755 $TMPFS_OPTIONS tmpfs "$tmpfs"`

Then, in ltsp.conf, under `[clients]`, put: `TMPFS_OPTIONS="-o size=10%"`

Then run `sudo ltsp initrd` and reboot the clients. Does it work that way?

**Reply by MaximShepelev** [reply]
Yes, this works. Thanks @alkisg. I guess this option can be added to main code.

```
ops@msh-pxe-test-3:~$ free
              total        used        free      shared  buff/cache   available
Mem:       10232768      162540     9158024      536584      912204     9286616
Swap:             0           0           0
ops@msh-pxe-test-3:~$ df 
Filesystem          1K-blocks   Used Available Use% Mounted on
udev                  5092448      0   5092448   0% /dev
tmpfs                 1023280   2244   1021036   1% /run
tmpfs                 1023280 534340    488940  53% /run/initramfs/ltsp
/dev/loop0             530048 530048         0 100% /run/initramfs/ltsp/0/looproot
/run/initramfs/ltsp   1023280 534340    488940  53% /
tmpfs                 5116384      0   5116384   0% /dev/shm
tmpfs                    5120      0      5120   0% /run/lock
tmpfs                 5116384      0   5116384   0% /sys/fs/cgroup
tmpfs                 1023276      0   1023276   0% /run/user/1000
```

Note to myself or anyone not reading full docs first: **ltsp.conf** file should be placed into **/etc/ltsp** before running `ltsp initrd`
